src/Plugin/search_api/processor/EntityReferenceWithUri.php

```diff
@@ -0,0 +1,195 @@
+      $container->get('islandora.utils'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+    if (!$datasource || !$datasource->getEntityTypeId()) {
+      return $properties;
+    }
+
+    $entity_type = $datasource->getEntityTypeId();
+
+    // Get all configured Entity Relation fields for this entity type.
+    $fields = \Drupal::entityTypeManager()->getStorage('field_config')->loadByProperties([
+      'entity_type' => $entity_type,
+      'field_type' => 'entity_reference',
+    ]);
+
+    foreach ($fields as $field) {
+      // Skip over fields that point to taxonomy term fields themselves.
+      if ($field->getSetting('target_type') == 'taxonomy_term') {
+        continue;
+      }
+
+      $definition = [
+        'label' => $this->t('@label (by term URI) [@bundle]', [
+          '@label' => $field->label(),
+          '@bundle' => $field->getTargetBundle(),
+        ]),
+        'description' => $this->t('Index the target entity, but only if the target entity has a taxonomy term.'),
+        'type' => 'string',
+        'processor_id' => $this->getPluginId(),
+        'settings' => [
+          'filter_terms' => [],
+          'target_type' => $field->getSetting('target_type'),
+        ],
+        'is_list' => TRUE,
+      ];
+      $fieldname = 'entity_reference_with_term__' . str_replace('.', '__', $field->id());
+      $properties[$fieldname] = new EntityReferenceWithUriProperty($definition);
+    }
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addFieldValues(ItemInterface $item) {
+    // Skip if no Entity Reference with URI fields are configured.
+    $relevant_search_api_fields = [];
+    $search_api_fields = $item->getFields(FALSE);
+    foreach ($search_api_fields as $field) {
+      if (substr($field->getPropertyPath(), 0, 28) == 'entity_reference_with_term__') {
+        $relevant_search_api_fields[] = $field;
+      }
+    }
+    if (empty($relevant_search_api_fields)) {
+      return;
+    }
+
+    $content_entity = $item->getOriginalObject()->getValue();
+    $type_and_bundle_prefix = $content_entity->getEntityTypeId() . '__' . $content_entity->bundle() . '__';
+    foreach ($relevant_search_api_fields as $search_api_field) {
+      $entity_type = $search_api_field->getConfiguration()['target_type'];
+      $filter_terms = $search_api_field->getConfiguration()['filter_terms'];
+      $filter_tids = array_map(function ($element) {
+        return $element['target_id'];
+      }, $filter_terms);
+      $require_all = $search_api_field->getConfiguration()['require_all'];
+      $field_name = substr($search_api_field->getPropertyPath(), 28);
+      $field_name = str_replace($type_and_bundle_prefix, '', $field_name);
+      if ($content_entity->hasField($field_name)) {
+        $referenced_terms = [];
+        foreach ($content_entity->get($field_name)->getValue() as $values) {
```

Comment:
Given that these are entity references can't this also take advantage of `referencedEntities()`?